### DISCUSSION TITLE
Automatic filetype detection for yaml.docker-compose

### DISCUSSION THREAD

**Discussion by robertmonka** [neovim/neovim#26571] [Q&A] [answered]
Hello.
How to configure that if I open file docker-compose.yml it will be detected as yaml.docker-compose?
By default it is recognised as yaml.

If I set manually :set filetype yaml.docker-compose, everything works fine.

**Comment by clason** [comment, marked as answer]
`:h vim.filetype.add()`

**Comment by robertmonka** [comment]
Thank you.
I added

vim.filetype.add({
  pattern = {
    ['docker-compose%.yml'] = 'yaml.docker-compose',
    ['docker-compose%.yaml'] = 'yaml.docker-compose',
    ['compose%.yml'] = 'yaml.docker-compose',
    ['compose%.yaml'] = 'yaml.docker-compose',
  },
})
to my init.lua file, but no effect at all.

**Reply by clason** [reply]
You want `filename`, I suspect. And try putting it in `.config/nvim/filetype.lua`.

**Comment by robertmonka** [comment]
Checked, but no effect at all.
Buffer detected as yaml not yaml.docker-compose

**Reply by clason** [reply]
Works for me. (If you use literal filenames, _don't_ escape the dot.)

**Comment by robertmonka** [comment]
Could you paste you file here?

**Reply by clason** [reply]
What file? I just put `vim.filetype.add({filename = { ['docker-compose.yaml'] = 'foo' }})` in my `init.lua` and did `nvim docker-compose.yaml`.

**Comment by robertmonka** [comment]
Now it works. Thank you

vim.filetype.add({
	filename = {
		["docker-compose.yml"] = "yaml.docker-compose",
		["docker-compose.yaml"] = "yaml.docker-compose",
		["compose.yml"] = "yaml.docker-compose",
		["compose.yaml"] = "yaml.docker-compose",
	},
})

**Comment by filipfigzalski** [comment]
I was trying to make it work with files named like `docker-compose.override.yml`. After struggling for some time with Lua patterns (which unfortunately don't support adding quantifiers to capture groups) I came up with the following solution:

```lua
vim.filetype.add({
    pattern = {
        ["compose.*%.ya?ml"] = "yaml.docker-compose",
        ["docker%-compose.*%.ya?ml"] = "yaml.docker-compose",
    },
})
```
